src/SparkTestBase.sol

```diff
@@ -161,16 +171,40 @@ abstract contract SpellRunner is Test {
             OptimismBridgeTesting.relayMessagesToDestination(bridge, false);
         } else if (bridgeType == BridgeType.CCTP) {
             CCTPBridgeTesting.relayMessagesToDestination(bridge, false);
+            CCTPBridgeTesting.relayMessagesToSource(bridge, false);
         } else if (bridgeType == BridgeType.GNOSIS) {
             AMBBridgeTesting.relayMessagesToDestination(bridge, false);
         }
     }
 
-    function executeL2PayloadsFromBridges() private onChain(ChainIdUtils.Ethereum()) {
-        // Base
-        chainSpellMetadata[ChainIdUtils.Base()].domain.selectFork();
-        IExecutor(Base.SPARK_EXECUTOR).execute(IExecutor(Base.SPARK_EXECUTOR).actionsSetCount() - 1);
-        // Gnosis: TODO
+    function _simulateExecuteForeignPayloads() private onChain(ChainIdUtils.Ethereum()) {
+        for (uint256 i = 0; i < allChains.length; i++) {
+            ChainId chainId = ChainIdUtils.fromDomain(chainSpellMetadata[allChains[i]].domain);
+            if (chainId == ChainIdUtils.Ethereum()) continue;  // Don't execute mainnet
+            if (_getMainnetPayloadFromSpell(chainId) != address(0)) continue;  // Payload is already defined in mainnet spell
+            address payload = chainSpellMetadata[chainId].payload;
+            if (payload != address(0)) {
+                // We will simulate execution until the real spell is deployed in the mainnet spell
+                chainSpellMetadata[chainId].domain.selectFork();
+                IExecutor executor = chainSpellMetadata[chainId].executor;
+                vm.prank(address(executor));
+                executor.executeDelegateCall(
+                    payload,
+                    abi.encodeWithSignature('execute()')
+                );
+            }
+        }
+    }
+
+    function _getMainnetPayloadFromSpell(ChainId chainId) internal onChain(ChainIdUtils.Ethereum()) returns (address) {
```

Comment:
this function name suggests it gets the mainnet payload address from the L2 one, when it does the contrary. Consider renaming it to `_getL2PayloadFromMainnetSpell` or something similar